src/analysis/souffle/tests/arcs_fact_tests/dynamic_rule_test.dl

```diff
@@ -16,48 +16,100 @@
 hasAppliedIntegrityTag("d", "defaultOwner", "Constant_1000").
 
-// This is the first policy rule.
+// Propagation rules for div_op
 mayHaveIntegrityTag(ap, owner, "TimestampSeconds") :-
   div_op(ap, operand1, operand2),
   mayHaveIntegrityTag(operand1, owner, "TimestampMillis"),
   mayHaveIntegrityTag(operand2, owner, "Constant_1000").
 
+// Default propagation rules.
+mayHaveIntegrityTag(ap, owner, tag) :-
+  div_op(ap, operand1, _),
+  mayHaveIntegrityTag(operand1, owner, tag),
+  tag != "TimestampMillis".
+
+mayHaveIntegrityTag(ap, owner, tag) :-
+  div_op(ap, _, operand2),
+  mayHaveIntegrityTag(operand2, owner, tag),
+  tag != "Constant_1000".
+
 // These rules propagating lacks of integrity tags across rules can only remove
 // integrity tags. Remove them for now to illustrate issue.
-// lacksIntegrityTag(ap, owner, "TimestampSeconds") :-
-//  div_op(ap, operand1, operand2),
-//  (lacksIntegrityTag(operand1, owner, "TimestampMillis");
-//   lacksIntegrityTag(operand2, owner, "Constant_1000")).
+lacksIntegrityTag(ap, owner, "TimestampSeconds") :-
+   div_op(ap, operand1, operand2),
+   (lacksIntegrityTag(operand1, owner, "TimestampMillis");
+    lacksIntegrityTag(operand2, owner, "Constant_1000")).
+
+// Default propagation rules.
+lacksIntegrityTag(ap, owner, tag) :-
+  div_op(ap, operand1, _),
+  lacksIntegrityTag(operand1, owner, tag),
+  tag != "TimestampMillis".
+
+lacksIntegrityTag(ap, owner, tag) :-
+  div_op(ap, _, operand2),
+  lacksIntegrityTag(operand2, owner, tag),
+  tag != "Constant_1000".
 
+//----------------------------------------
 // This is the second policy rule.
+// Propagation rules for mult_op.
 mayHaveIntegrityTag(ap, owner, "TimestampMillisRedacted") :-
   mult_op(ap, operand1, operand2),
   mayHaveIntegrityTag(operand1, owner, "TimestampSeconds"),
-  mayHaveIntegrityTag(operand1, owner, "Constant_1000").
+  mayHaveIntegrityTag(operand2, owner, "Constant_1000").
+
+// Default propagation rules.
+mayHaveIntegrityTag(ap, owner, tag) :-
+  mult_op(ap, operand1, _),
+  mayHaveIntegrityTag(operand1, owner, tag),
+  tag != "TimestampSeconds".
+
+mayHaveIntegrityTag(ap, owner, tag) :-
+  mult_op(ap, _, operand2),
+  mayHaveIntegrityTag(operand2, owner, tag),
+  tag != "Constant_1000".
 
-// lacksIntegrityTag(ap, owner, "TimestampMillisRedacted") :-
-//  mult_op(ap, operand1, operand2),
-//  (lacksIntegrityTag(operand1, owner, "TimestampSeconds");
-//   lacksIntegrityTag(operand2, owner, "Constant_1000")).
+lacksIntegrityTag(ap, owner, "TimestampMillisRedacted") :-
+ mult_op(ap, operand1, operand2),
+ (lacksIntegrityTag(operand1, owner, "TimestampSeconds");
+  lacksIntegrityTag(operand2, owner, "Constant_1000")).
+
+// Default propagation rules.
+lacksIntegrityTag(ap, owner, tag) :-
+  mult_op(ap, operand1, _),
+  lacksIntegrityTag(operand1, owner, tag),
+  tag != "TimestampSeconds".
+
+lacksIntegrityTag(ap, owner, tag) :-
+  mult_op(ap, _, operand2),
+  lacksIntegrityTag(operand2, owner, tag),
+  tag != "Constant_1000".
```

Comment:
Also suspicious of these default propagation rules excluding `Constant_1000` as above.